### PR TITLE
Helm(hooks): honor imagePullPolicy and imagePullSecrets

### PR DESCRIPTION
**Pull request by benashz**
Update all Helm hook Jobs to Honor the the manager's `imagePullPolicy` and `imagePullSecrets`.

### CODE REVIEW

File: chart/templates/deployment.yaml [hunk 1]
```diff
@@ -194,9 +194,14 @@ spec:
       serviceAccountName: {{ include "vso.chart.fullname" . }}-controller-manager
       securityContext:
         {{- toYaml .Values.controller.podSecurityContext | nindent 8 }}
+      {{- with .Values.controller.imagePullSecrets }}
+      imagePullSecrets:
+      {{- toYaml . | nindent 8 }}
+      {{- end }}
```
Comment (tvoran) [inline chart/templates/deployment.yaml:200]: The image pull secrets are already set on the deployment's service account (line 14), so I don't think this is necessary.

File: chart/templates/hook-upgrade-crds.yaml [hunk 2]
```diff
@@ -88,9 +88,14 @@ spec:
       serviceAccountName: {{ template "vso.chart.fullname" . }}-upgrade-crds
       securityContext:
         {{- toYaml .Values.controller.podSecurityContext | nindent 8 }}
+      {{- with .Values.controller.imagePullSecrets }}
+      imagePullSecrets:
+      {{- toYaml . | nindent 8 }}
+      {{- end }}
```
Comment (tvoran) [inline chart/templates/hook-upgrade-crds.yaml:94]: How about setting the pull secrets on the service account, like we do for [the deployment's service account](https://github.com/hashicorp/vault-secrets-operator/blob/be6ff476c51d98aa78b3290e62ac3271813df38e/chart/templates/deployment.yaml#L14)?

i.e.

```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: {{ template "vso.chart.fullname" . }}-upgrade-crds
  namespace: {{ .Release.Namespace }}
  labels:
    app.kubernetes.io/component: controller-manager
{{ include "vso.chart.labels" . | indent 4 }}
  annotations:
    helm.sh/hook: pre-upgrade
    helm.sh/hook-delete-policy: "hook-succeeded,before-hook-creation"
    helm.sh/hook-weight: "1"
{{ include "vso.imagePullSecrets" .}}

File: chart/templates/deployment.yaml (hunk 1 above)
Comment (benashz) [reply chart/templates/deployment.yaml]: Reverted in c1dfa0095f554f9f65e94605cbcbf0e2ea0b71a3

File: chart/templates/hook-upgrade-crds.yaml (hunk 2 above)
Comment (benashz) [reply chart/templates/hook-upgrade-crds.yaml]: 👍 updated in c1dfa0095f554f9f65e94605cbcbf0e2ea0b71a3

**Review by tvoran — APPROVED** [review]

**Pull request merged**